haiku-misc/haiku-on/haiku_on_x86-1.0.recipe

```diff
@@ -0,0 +1,51 @@
+SUMMARY="A project to create language bindings for the Haiku API"
+DESCRIPTION="Haiku On is a project to create language bindings \
+for the Haiku API. The first language to be targeted is Java, \
+and it will likely be the only targeted language for quite some \
+time. However, with Java bindings come Jython, JRuby, Scala etc. \
+bindings, so we can get a really good return from targeting Java."
+HOMEPAGE="https://github.com/yourpalal/Haiku-On"
+COPYRIGHT="2012 Alex Wilson"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="48834fe631b4054fe132bf9b7498db222f40f182"
+SOURCE_URI="https://github.com/yourpalal/Haiku-On/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="7f10e65da03d04b0c4c245cbfe0fa88ee2b301f282a9562a9fb38c4548d509bc"
+SOURCE_FILENAME="Haiku-On-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="Haiku-On-$srcGitRev"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	haiku_on$secondaryArchSuffix = $portVersion
+	lib:libhaiku_onJava.so$secondaryArchSuffix = $portVersion
+	cmd:haiku.jar = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	cmd:swig$secondaryArchSuffix
+	cmd:openjdk$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku_devel$secondaryArchSuffix
+	cmd:swig$secondaryArchSuffix
+	cmd:openjdk$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD()
+{
+	ln -s ../misc/ecj $binDir
+	ln -s `which gjar` $binDir/jar
```

Comment:
Did you check if you could create a symbolic link here?

Reply:
No, Do I need to check it?

Reply:
It would help finding out if you tried building it yourself :) (in case there are errors you probably need to fix  those then)

Reply:
@Begasus How can I build it if openjdk isn't on haiku :P

Reply:
@arnavbhatt288 this doensn't involve openjdk yet ;) (ps, there is a package for openjdk floating around, but had no luck with it, it provides jar (so that shouldn't be linked) but it doesn't include ecj, also as said earlier, it can't seem to find the header(s) jni.h

Reply:
> @arnavbhatt288 this doensn't involve openjdk yet ;) (ps, there is a package for openjdk floating around, but had no luck with it, it provides jar (so that shouldn't be linked) but it doesn't include ecj, also as said earlier, it can't seem to find the header(s) jni.h

@Begasus It does, it doesn't show errors for any permission.